pkg/plugin/aggregation/update_test.go

```diff
@@ -210,3 +211,114 @@ func TestGetAggregatorPodName(t *testing.T) {
 		})
 	}
 }
+
+func TestReceive(t *testing.T) {
+	expectedResults := []plugin.ExpectedResult{
+		{NodeName: "node1", ResultType: "type1"},
+		{NodeName: "node2", ResultType: "type1"},
+		{NodeName: "global", ResultType: "type2"},
+	}
+	tcs := []struct {
+		desc     string
+		results  map[string]*plugin.Result
+		updates  map[string]*plugin.StatusUpdate
+		expected *Status
+	}{
+		{
+			desc: "No results or updates",
+			expected: &Status{
+				Plugins: []PluginStatus{
+					{Node: "node1", Plugin: "type1", Status: "running"},
+					{Node: "node2", Plugin: "type1", Status: "running"},
+					{Node: "global", Plugin: "type2", Status: "running"},
+				},
+				Status: "running",
+			},
+		}, {
+			desc: "Status updates without results",
+			expected: &Status{
+				Plugins: []PluginStatus{
+					{
+						Node: "node1", Plugin: "type1", Status: "running",
+						Progress: &plugin.StatusUpdate{
+							PluginName: "type1", Node: "node1", Status: "new status",
+						},
+					},
+					{Node: "node2", Plugin: "type1", Status: "running"},
+					{Node: "global", Plugin: "type2", Status: "running"},
+				},
+				Status: "running",
+			},
+			updates: map[string]*plugin.StatusUpdate{
+				"type1/node1": {PluginName: "type1", Node: "node1", Status: "new status"},
+			},
+		}, {
+			desc: "Results without status updates",
+			expected: &Status{
+				Plugins: []PluginStatus{
+					{
+						Node: "node1", Plugin: "type1", Status: "complete",
+					},
+					{Node: "node2", Plugin: "type1", Status: "running"},
+					{Node: "global", Plugin: "type2", Status: "running"},
+				},
+				Status: "running",
+			},
+			results: map[string]*plugin.Result{
+				"type1/node1": {NodeName: "node1", ResultType: "type1"},
+			},
+		}, {
+			desc: "Overlapping status updates and results",
+			expected: &Status{
+				Plugins: []PluginStatus{
+					{
+						Node: "node1", Plugin: "type1", Status: "complete",
+						Progress: &plugin.StatusUpdate{
+							PluginName: "type1", Node: "node1", Status: "new status",
+						},
+					},
+					{Node: "node2", Plugin: "type1", Status: "running"},
+					{Node: "global", Plugin: "type2", Status: "running"},
+				},
+				Status: "running",
+			},
+			results: map[string]*plugin.Result{
+				"type1/node1": {NodeName: "node1", ResultType: "type1"},
+			},
+			updates: map[string]*plugin.StatusUpdate{
+				"type1/node1": {PluginName: "type1", Node: "node1", Status: "new status"},
+			},
+		}, {
+			desc: "Non-overlapping updates and results",
```

Comment:
TODO: 
 - [ ] test with failing result
 - [ ] test with failures and errors in the status update

Reply:
Glad I did these; found a bug in the code that prevented some data from being reported.